programs/steward/src/state/steward_state.rs

```diff
@@ -39,8 +42,13 @@ pub fn maybe_transition_and_emit(
     params: &Parameters,
     epoch_schedule: &EpochSchedule,
 ) -> Result<()> {
+    if clock.epoch != state_account.current_epoch {
```

Comment:
Better to have this as a `require!()` at the beginning of all state machine instructions rather than hit this function and fail at the end, less CUs used and more clear

Reply:
Done

Reply:
Done